datasheet/Zubax_GNSS2_Datasheet.tex

```diff
@@ -0,0 +1,777 @@
+
+\begin{ZubaxSimpleTable}{Bootloader states and UAVCAN node status codes}{| X | X | X |}
+Bootloader state & UAVCAN node mode & UAVCAN node health\\
+NoAppToBoot & SOFTWARE{\_}UPDATE & ERROR\\
+BootDelay, ReadyToBoot & MAINTENANCE & OK\\
+BootCancelled & MAINTENANCE & WARNING\\
+AppUpgradeInProgress & SOFTWARE{\_}UPDATE & OK
+\end{ZubaxSimpleTable}
+
+The vendor specific status code of the node status message contains the status code of the last attempt to upgrade the firmware. Please refer to the error code table provided above.
+
+\clearpage
+\subsection{Supported messages}
+\begin{ZubaxSimpleTable}{Supported messages}{| l | l | l | l |}
+Data type & Direction & Period & Transfer priority \\
+\texttt{uavcan.protocol.NodeStatus} & Output & 1 Hz & 24(low) \\
+\texttt{uavcan.protocol.debug.LogMessage} & Output & Aperiodic & 31(lowest) \\
+\texttt{uavcan.protocol.dynamic{\_}node{\_}id.Allocation} & Input/Output & Aperiodic & 24 (low)
+\end{ZubaxSimpleTable}
+
+\begin{ZubaxSimpleTable}{Supported messages annotation}{| l | X |}
+Data type & Note\\
+\texttt{uavcan.protocol.NodeStatus} & Refer to the bootloader state mapping table.\\
+\texttt{uavcan.protocol.debug.LogMessage} & Used to report the application upgrade progress and status in a human readable form.\\
+\texttt{uavcan.protocol.dynamic{\_}node{\_}id.Allocation} & Used only during initialization, if the application did not provide a specific node ID to use.
+\end{ZubaxSimpleTable}
+\clearpage
+\subsection{Supported services}
+
+Servers: 
+
+\begin{ZubaxSimpleTable}{Supported servers}{| l | X |}
+Data type & Note\\
+uavcan.protocol.GetNodeInfo & The nested structure  \textbf{uavcan.protocol.SoftwareVersion} is populated only if there is a valid application.\\
+uavcan.protocol.RestartNode & Restarts the bootloader.\\
+uavcan.protocol.file.BeginFirmwareUpdate & Initiates the firmware update process, unless it was initiated earlier before the application has rebooted into the bootloader.
+\end{ZubaxSimpleTable}
+
+Clients:
+
+\begin{ZubaxSimpleTable}{Supported clients}{| l | l | l | X |}
+Data type & Response timeout & Transfer priority & Note \\
+uavcan.protocol.file.Read & 1 second & 24(low) & Used to download the application image file from the specified file server.
+\end{ZubaxSimpleTable}
+
+The interval at which the file read requests are issued while downloading the application image is defined by the following formula (all units SI):
+
+\textbf{last{\_}response{\_}latency + 1 / (1 + can{\_}bus{\_}bit{\_}rate / 65536)}
+
+This formula, combined with the low transfer priority, allows the bootloader to avoid congestion of the CAN bus while downloading the firmware image.
+
+\chapter{Mechanical characteristics}
+The drawing below documents basic mechanical  characteristics of Zubax GNSS 2, such as the
+placement of connectors and mounting holes.
+\begin{figure}[!hbt]
+	\centerline{\includegraphics[width=1.1\textwidth]{GNSS2_drawing}}
```

Comment:
The drawing renders way too small. I suspect the image has large margins on both sides, they should be removed.